### ISSUE TITLE
no-invalid-this fails when used with ES7 class property declaration

### ISSUE THREAD

**Issue by gajus** [babel/babel-eslint#173]
```js
class {
    toggleLocationActiveState = () => {
        this.props.onActivateLocation(this.props.id);
    }
}
```

```
curiosity:foo gajus$ eslint -v
v1.2.1
```

```
/foo/index.js
  13:9   error  Unexpected `this`                       no-invalid-this
  13:39  error  Unexpected `this`                       no-invalid-this
```

https://github.com/eslint/eslint/issues/3475

**Comment by gajus** [comment]
`"babel-eslint": "^4.0.10",`

**Comment by hzoo** [comment]
I'l have to look into it more but most likely the rule will need to be added to eslint-plugin-babel

**Comment by hzoo** [comment]
Yeah looking at the rule source that seems like the only way to fix it - I would file it there (https://github.com/babel/eslint-plugin-babel)
